chia/pools/pool_wallet.py

```diff
@@ -462,14 +463,24 @@ async def create_new_pool_wallet_transaction(
         )
         return standard_wallet_record, p2_singleton_puzzle_hash, launcher_coin_id
 
-    async def sign(self, coin_spend: CoinSpend) -> SpendBundle:
+    async def sign(self, coin_spend: CoinSpend, fee: uint64) -> SpendBundle:
         async def pk_to_sk(pk: G1Element) -> PrivateKey:
             owner_sk: Optional[PrivateKey] = await find_owner_sk([self.wallet_state_manager.private_key], pk)
-            assert owner_sk is not None
+            # assert owner_sk is not None
+            if owner_sk is None:
+                return self.standard_wallet.secret_key_store.secret_key_for_public_key(pk)
```

Comment:
This line is a hack to get the standard wallet key for signing.

Probably better to do this with `SpendBundle.aggregate`, as below

Reply:
Why can't we call `generate_signed_transaction`?

Reply:
Good idea.